docs/benchmarking.md

```
@@ -0,0 +1,252 @@
+
+- **>5%** for any core operation without justification
+- **>10%** for any benchmark (indicates algorithmic issue)
+- **Any regression** in AST parsing (critical path)
+
+## CI Integration
+
+### Bencher.dev Integration
+
+Every PR automatically receives benchmark comparison comments with visual charts and statistical analysis:
+
+```
```

markdownlint-cli2 (0.17.2)

Fenced code blocks should have a language specified
null

(MD040, fenced-code-language)